### ISSUE TITLE
UX for failing/failed downloads confused me

### ISSUE THREAD

**Issue by julianharty** [kiwix/kiwix-android#324]
There seem to be aspects of the current UI that don't communicate what's happening clearly. 

For instance, the following content is not available from the server. However rather than reporting an error to the user, the GUI remains as follows (ignoring for now the red box highlighting the relevant part of the GUI). 

<img width="344" alt="kiwix_downloading_ui_for_missing_item" src="https://user-images.githubusercontent.com/785891/34016728-e236412a-e11a-11e7-8015-b7c694149b70.png">

The app crashes after pausing and resuming the download. See stacktrace for details:
```
12-14 22:07:31.379 10498 10591 D OkHttp  : --> GET http://kiwix-download-server/zim/tinytest.zim http/1.1
12-14 22:07:31.391 10498 10591 D OkHttp  : <-- 404 Not Found http://kiwix-download-server/zim/tinytest.zim (11ms, unknown-length body)
12-14 22:07:33.417 10498 10498 I kiwixdownloadservice: Pausing ZIM Download for notificationID: 48696
12-14 22:07:33.428 10498 10498 D AndroidRuntime: Shutting down VM
12-14 22:07:33.428 10498 10498 E AndroidRuntime: FATAL EXCEPTION: main
12-14 22:07:33.428 10498 10498 E AndroidRuntime: Process: org.kiwix.kiwixmobile, PID: 10498
12-14 22:07:33.428 10498 10498 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.ListView.invalidateViews()' on a null object reference
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at org.kiwix.kiwixmobile.downloader.DownloadService.pauseDownload(DownloadService.java:235)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at org.kiwix.kiwixmobile.downloader.DownloadFragment$DownloadAdapter.setPlayState(DownloadFragment.java:201)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at org.kiwix.kiwixmobile.downloader.DownloadFragment$DownloadAdapter.lambda$getView$2$DownloadFragment$DownloadAdapter(DownloadFragment.java:251)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at org.kiwix.kiwixmobile.downloader.DownloadFragment$DownloadAdapter$$Lambda$1.onClick(Unknown Source)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at android.view.View.performClick(View.java:5204)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at android.view.View$PerformClick.run(View.java:21153)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:739)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:95)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:148)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5417)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
12-14 22:07:33.428 10498 10498 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
12-14 22:07:33.430   650   932 W ActivityManager:   Force finishing activity org.kiwix.kiwixmobile/.zim_manager.ZimManageActivity
12-14 22:07:33.482   650   729 I Adreno-EGL: <qeglDrvAPI_eglInitialize:379>: QUALCOMM Build: 10/21/15, 369a2ea, I96aee987eb
```
The file being downloaded is not listed in the `ONLINE` tab. I presume that's because we don't want users to try downloading it several times, however it's a bit perturbing to discover it's no longer available online. 

We may be able to get some help and advice from a UX guru. For the moment I think it's safe to assume we can improve the error reporting when the file isn't available to be downloaded and also to stop the Activity from crashing.

**Comment by kelson42** [comment]
@julianharty @macgills Do I'm right or this bug has been fixed/deprecated by 3.0?

**Comment by macgills** [comment]
would have been fixed for 2.5.X
